services/quotas-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/QuotasServiceDAL.java

```diff
@@ -367,6 +369,77 @@ private Map<String, Double> getQuotas(String serviceCode) {
         }
     }
 
+    // AWS Service Quotas is currently unavailable in the GCR regions, so we use the quota from service itself for check.
+    public QuotaCheck checkQuotasForCNRegion() {
+        String serviceCode;
+        Map<String, Double> deployedCountMap = new LinkedHashMap<>();
+        Map<String, Double> quotasMap = new LinkedHashMap<>();
+        StringBuilder builder = new StringBuilder();
+
+        boolean reportBackError = false;
+        boolean exceedsLimit = false;
+        List<Service> retList = new ArrayList<>();
+        // RDS
+        serviceCode = "rds";
+        deployedCountMap.clear();
+        deployedCountMap.put("DB clusters", Double.valueOf(getRdsClusters()));
+        deployedCountMap.put("DB instances", Double.valueOf(getRdsInstances()));
+        quotasMap = getRdsInstancesQuota();
+        exceedsLimit = compareValues(retList, deployedCountMap, serviceCode, quotasMap, builder);
+        reportBackError = reportBackError || exceedsLimit;
+
+        // load balancers
+        serviceCode = "elasticloadbalancing";
+        deployedCountMap.clear();
+        deployedCountMap.put("Application Load Balancers per Region", Double.valueOf(getAlbs()));
+        quotasMap = getELBQuota();
+        exceedsLimit = compareValues(retList, deployedCountMap, serviceCode, quotasMap, builder);
+        reportBackError = reportBackError || exceedsLimit;
+
+        QuotaCheck quotaCheck = new QuotaCheck();
+        quotaCheck.setPassed(!reportBackError);
+        quotaCheck.setServiceList(retList);
+        quotaCheck.setMessage(builder.toString());
+        return quotaCheck;
+    }
+
+    private Map<String, Double> getRdsInstancesQuota() {
+        long instances;
+        long clusters;
+        try {
+            clusters = rds.describeAccountAttributes().accountQuotas().stream()
+                    .filter(quota -> quota.accountQuotaName().equals("DBClusters"))
+                    .findFirst().map(AccountQuota::max).orElse(40L);
+            instances = rds.describeAccountAttributes().accountQuotas().stream()
+                    .filter(quota -> quota.accountQuotaName().equals("DBInstances"))
+                    .findFirst().map(AccountQuota::max).orElse(40L);
```

Comment:
Let's cache the result of `rds.describeAccountAttributes().accountQuotas` so we don't waste time doing a second round-trip call to RDS.

Reply:
I think if we cache this result, then account changes its quota, we may use the old value.

Reply:
I use a accountQuotas field to cache the result.